.travis.yml

```diff
@@ -1,21 +1,37 @@
+sudo: false
 language: python
 
-python:
-  - "2.7"
-  - "3.3"
-  - "3.4"
-  - "3.5"
-  - "pypy-5.3.1"
+cache:
+  directories:
+    - $HOME/.cache/pip
+    - .hypothesis
+
+matrix:
+  include:
+    # Main tests on supported Python versions.
+    - python: "2.7"
+      env: TOXENV=py27
+    - python: "3.3"
+      env: TOXENV=py33
+    - python: "3.4"
+      env: TOXENV=py34
+    - python: "3.5"
+      env: TOXENV=py35
+    - python: "pypy-5.3.1"
```

Comment:
Also, Travis appears to have PyPy 5.4 available, just not 5.4.1 (yet). Is that preferable to 5.3.1?

https://github.com/travis-ci/travis-ci/issues/6553